### PR TITLE
Io add custom vertexattribs

### PR DESCRIPTION
**Pull request by nmellado**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Allow to load vertex attributes from a ply file

* **What is the current behavior?** (You can also link to an open issue here)

The ply loader only loads "standard" vertex attributes : position, normal and color

* **What is the new behavior (if this is a feature change)?**

The ply loader also loads all the vertex attributes defined in the ply file

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

No

* **Other information**:

### CODE REVIEW

**Comment by nmellado** [comment]
Example of file to be loaded:  [goursat_0100000APSS0.ply.zip](https://github.com/STORM-IRIT/Radium-Engine/files/6234703/goursat_0100000APSS0.ply.zip)

Next step: use https://github.com/nmwsharp/happly

**Comment by MathiasPaulin** [comment]
> Example of file to be loaded: [goursat_0100000APSS0.ply.zip](https://github.com/STORM-IRIT/Radium-Engine/files/6234703/goursat_0100000APSS0.ply.zip)

The crash is fixed and this file loads well currently.


> Next step: use https://github.com/nmwsharp/happly
Is this really needed now ?

**Review by nmellado — COMMENTED** [review]
Fine for me.
@dlyr can you please review this PR, we need it for a project.
Thanks

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp [hunk 1]
```diff
@@ -74,6 +75,66 @@ struct memory_stream : virtual memory_buffer, public std::istream {
         memory_buffer( first_elem, size_ ), std::istream( static_cast<std::streambuf*>( this ) ) {}
 };
 
+template <typename DataType, typename ContainerType>
+void copyArrayToRadiumcontainer( const uint8_t* buffer, ContainerType& container, size_t count ) {
```
Comment (dlyr) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp:79]:
```suggestion
void copyArrayToContainer( const uint8_t* buffer, ContainerType& container, size_t count ) {
```
I don't get why the container is "Radium" here

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp [hunk 2]
```diff
@@ -74,6 +75,66 @@ struct memory_stream : virtual memory_buffer, public std::istream {
...
+    CORE_UNUSED( buffer );
```
Comment (dlyr) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp:80]: Could we avoid these and simple comment or remove args names ?

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp [hunk 3]
```diff
@@ -74,6 +75,66 @@ struct memory_stream : virtual memory_buffer, public std::istream {
...
+    CORE_UNUSED( container );
+    CORE_UNUSED( count );
+    LOG( logERROR )
```
Comment (dlyr) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp:83]: Should this error be raised at compile time ? I think so (so use a static_assert for instance)

File: src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp [hunk 4]
```diff
@@ -174,29 +221,49 @@ FileData* TinyPlyFileLoader::loadFile( const std::string& filename ) {
...
-            container.reserve( buffer->count );
-            for ( size_t i = 0; i < buffer->count; ++i )
+            for ( const auto& p : e.properties )
             {
-                container.emplace_back(
-                    floatBuffer[i * 3 + 0], floatBuffer[i * 3 + 1], floatBuffer[i * 3 + 2] );
+                bool exists = usedAttributes.find( p.name ) != usedAttributes.end();
+                if ( !exists )
+                {
+#ifdef OUTPUT_CUSTOM_ATTRIBS
```
Comment (dlyr) [inline src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp:252]: could we get rid of this define ?

**Review by dlyr — COMMENTED** [review]
First round of review. 
I do not get at a first read what is not working yet on master, and how it is handled in this PR.

A test file that fail on current master and now is handled correclty is welcom, or a bit of explanation of what is fixed.
My first guess is that you want to load custom attributes from .ply file to core mesh attribs.

**Review by dlyr — APPROVED** [review]

**Pull request merged**